apiserver/src/main/resources/migration/changelog-v5.6.0.xml

```diff
@@ -1430,4 +1430,201 @@
+            ALTER INDEX "PROJECTMETRICS_NEW_FIRST_OCCURRENCE_IDX" RENAME TO "PROJECTMETRICS_FIRST_OCCURRENCE_IDX";
+            ALTER INDEX "PROJECTMETRICS_NEW_LAST_OCCURRENCE_IDX" RENAME TO "PROJECTMETRICS_LAST_OCCURRENCE_IDX";
+        </sql>
+        <!--Migration for "DEPENDENCYMETRICS"-->
+        <sql>
+            -- create new partitioned table with new PK
+            CREATE TABLE "DEPENDENCYMETRICS_NEW"
+            (
+                "ID" BIGSERIAL,
+                "COMPONENT_ID" int8 NOT NULL,
+                "CRITICAL" int4 NOT NULL,
+                "FINDINGS_AUDITED" int4 NULL,
+                "FINDINGS_TOTAL" int4 NULL,
+                "FINDINGS_UNAUDITED" int4 NULL,
+                "FIRST_OCCURRENCE" timestamptz NOT NULL,
+                "HIGH" int4 NOT NULL,
+                "RISKSCORE" float8 NOT NULL,
+                "LAST_OCCURRENCE" timestamptz NOT NULL,
+                "LOW" int4 NOT NULL,
+                "MEDIUM" int4 NOT NULL,
+                "POLICYVIOLATIONS_AUDITED" int4 NULL,
+                "POLICYVIOLATIONS_FAIL" int4 NULL,
+                "POLICYVIOLATIONS_INFO" int4 NULL,
+                "POLICYVIOLATIONS_LICENSE_AUDITED" int4 NULL,
+                "POLICYVIOLATIONS_LICENSE_TOTAL" int4 NULL,
+                "POLICYVIOLATIONS_LICENSE_UNAUDITED" int4 NULL,
+                "POLICYVIOLATIONS_OPERATIONAL_AUDITED" int4 NULL,
+                "POLICYVIOLATIONS_OPERATIONAL_TOTAL" int4 NULL,
+                "POLICYVIOLATIONS_OPERATIONAL_UNAUDITED" int4 NULL,
+                "POLICYVIOLATIONS_SECURITY_AUDITED" int4 NULL,
+                "POLICYVIOLATIONS_SECURITY_TOTAL" int4 NULL,
+                "POLICYVIOLATIONS_SECURITY_UNAUDITED" int4 NULL,
+                "POLICYVIOLATIONS_TOTAL" int4 NULL,
+                "POLICYVIOLATIONS_UNAUDITED" int4 NULL,
+                "POLICYVIOLATIONS_WARN" int4 NULL,
+                "PROJECT_ID" int8 NOT NULL,
+                "SUPPRESSED" int4 NOT NULL,
+                "UNASSIGNED_SEVERITY" int4 NULL,
+                "VULNERABILITIES" int4 NOT NULL,
+                CONSTRAINT "DEPENDENCYMETRICS_NEW_PK" PRIMARY KEY ("ID", "LAST_OCCURRENCE")
+            ) PARTITION BY RANGE ("LAST_OCCURRENCE");
+
+            -- create new indexes
+            CREATE INDEX "DEPENDENCYMETRICS_NEW_FIRST_OCCURRENCE_IDX" ON "DEPENDENCYMETRICS_NEW" ("FIRST_OCCURRENCE");
+            CREATE INDEX "DEPENDENCYMETRICS_NEW_LAST_OCCURRENCE_IDX" ON "DEPENDENCYMETRICS_NEW" ("LAST_OCCURRENCE");
+
+            -- call the migration procedure
+            CALL "MIGRATE_METRICS_TO_PARTITIONS"('DEPENDENCYMETRICS_NEW'::TEXT, 'DEPENDENCYMETRICS'::TEXT);
+
+            -- drop old table
+            DROP TABLE "DEPENDENCYMETRICS";
+
+            -- rename partitioned table
+            ALTER TABLE "DEPENDENCYMETRICS_NEW" RENAME TO "DEPENDENCYMETRICS";
+
+            -- rename PK
+            ALTER TABLE "DEPENDENCYMETRICS" RENAME CONSTRAINT "DEPENDENCYMETRICS_NEW_PK" TO "DEPENDENCYMETRICS_PK";
+
```

Comment:
The `ID` column creates a sequence behind the scenes, which we'll need to rename as well:

```sql
ALTER SEQUENCE "DEPENDENCYMETRICS_NEW_ID_seq" RENAME TO "DEPENDENCYMETRICS_ID_seq";
```